src/obfl-specification.html

```diff
@@ -1167,7 +1167,7 @@ <h4 id="L1023" class="include">block</h4>
       </dl>
     </dd>
   <dt>Content Model</dt>
-    <dd>text(), leader, marker, br, evaluate, page-number, marker-reference,
+    <dd>text(), leader, marker, br, evaluate, page-number, blockmeta, marker-reference,
```

Comment:
The `text()`, `leader`, `marker`, `br`, etc. are so-called "block contents". If you allow an `external-reference` here, you must also allow it _everywhere_. That is, all these elements should have `external-reference` in their Content Model:
```
before
page-area
block
td
toc-entry
toc-entry-on-resumed
volume-transition
block-resumed
item
```

Reply:
No, the content model for a block don't have to be equal to other similar types of elements that are block-like. Having external-references in a table of contents is not necessarily something that we want.

Then again for use-cases that @bertfrees have pointed out, it could be something we want to add in the future.

Reply:
@kalaspuffar is right, however I think @PaulRambags has a good point that it would bring consistency in the spec if you also change the content model of these other elements. You don't even need to support it in Dotify at this point, although it would be trivial (if it's not already supported).

Reply:
To extend it to elements like `field` is also a possibility although a slightly different story because it might require some more work in the implementation.

Reply:
Added to the content model of blocks below.
```
before
block
td
toc-entry
toc-entry-on-resumed
block-resumed
item
field
```

I did not add it to these because they are ordered in a specific order which and the reference doesn't have a good place to be in the order.
```
volume-transition
page-area
```

Reply:
> I did not add it to these because they are ordered in a specific order which and the reference doesn't have a good place to be in the order.

What do you mean exactly?

Oh by the way there is also `after`, which Paul didn't mention. And there might be other elements, I haven't checked.

Reply:
Added external-reference on all places we use marker-reference.

What I meant was that page-area and volume-transition as elements don't need a direct child of external-reference. They have particular children in order. Some of the children can have references so you should be able to put the reference in at a suggested position.

Reply:
Ah yes, of course.